kubernetes_asyncio/stream/ws_client.py

```diff
@@ -91,9 +107,14 @@ async def request(self, method, url, query_params=None, headers=None,
                         if data:
                             if channel in [STDOUT_CHANNEL, STDERR_CHANNEL]:
                                 resp_all += data
+                            elif channel == ERROR_CHANNEL:
+                                error_data += data
+
+            if error_data:
+                self.parse_error_data(error_data)
 
             return WsResponse(200, resp_all.encode('utf-8'))
 
         else:
 
-            return await self.rest_client.pool_manager.ws_connect(url, headers=headers, heartbeat=self.heartbeat)
+            return self.rest_client.pool_manager.ws_connect(url, headers=headers, heartbeat=self.heartbeat)
```

Comment:
Yes, it's a breaking change. I agree that a context manager would be helpful, so we can do it but it'll be released in a month or so (version 31.x, K8s v1.31 is not released yet).

Reply:
Sure, do what you feel the most relevant for this patch. On my side I can easily adapt my code.